RockLib.Messaging.RabbitMQ.Tests/RabbitSenderTests.cs

```diff
@@ -0,0 +1,207 @@
+﻿using FluentAssertions;
+using Moq;
+using RabbitMQ.Client;
+using System;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace RockLib.Messaging.RabbitMQ.Tests
+{
+    public static class RabbitSenderTests
+    {
+        [Fact]
+        public static void Create()
+        {
+            var model = new Mock<IModel>(MockBehavior.Strict);
+            model.Setup(_ => _.Dispose());
+
+            var connection = new Mock<IConnection>(MockBehavior.Strict);
+            connection.Setup(_ => _.CreateModel()).Returns(model.Object);
+            connection.Setup(_ => _.Dispose());
+
+            var connectionFactory = new Mock<IConnectionFactory>(MockBehavior.Strict);
+            connectionFactory.Setup(_ => _.CreateConnection()).Returns(connection.Object);
+
+            using (var sender = new RabbitSender("name", connectionFactory.Object,
+                "exchange", "routingKey", "routingKeyHeaderName", false))
+            {
+                sender.Name.Should().Be("name");
+                sender.Exchange.Should().Be("exchange");
+                sender.RoutingKey.Should().Be("routingKey");
+                sender.RoutingKeyHeaderName.Should().Be("routingKeyHeaderName");
+                sender.Persistent.Should().Be(false);
+                sender.Connection.Should().Be(connection.Object);
+                sender.Channel.Should().Be(model.Object);
+            }
+
+            connectionFactory.VerifyAll();
+            connection.VerifyAll();
+            model.VerifyAll();
+        }
+
+        [Fact]
+        public static void DisposeWhenChannelAndConnectionAreNotCreated()
+        {
+            var connectionFactory = new Mock<IConnectionFactory>(MockBehavior.Strict);
+
+            using (var sender = new RabbitSender("name", connectionFactory.Object,
+                "exchange", "routingKey", "routingKeyHeaderName", false))
+            { }
+
+            connectionFactory.VerifyAll();
+        }
+
+        [Fact]
+        public static void DisposeWhenChannelIsNotCreated()
+        {
+            var connection = new Mock<IConnection>(MockBehavior.Strict);
+            connection.Setup(_ => _.Dispose());
+
+            var connectionFactory = new Mock<IConnectionFactory>(MockBehavior.Strict);
+            connectionFactory.Setup(_ => _.CreateConnection()).Returns(connection.Object);
+
+            using (var sender = new RabbitSender("name", connectionFactory.Object,
+                "exchange", "routingKey", "routingKeyHeaderName", false))
+            {
+                _ = sender.Connection;
```

Comment:
What does this give us?  Not understanding why the assignment to a variable we don't care about.

Reply:
It's a little weird, but there's a `Lazy<IConnection>` underneath the scenes, so to get the disposal logic tested in the right way, I need to read the property, but I don't need to do anything with it. Hence the use of the discard.

Reply:
👍   Cool figured I was missing something.